lib/src/directus_api_manager_base.dart

```diff
@@ -205,76 +205,94 @@ class DirectusApiManager {
-      required Map<String, dynamic> objectData,
-      required Type Function(dynamic) jsonConverter}) {
+  Future<DirectusItemCreationResult<Type>>
+      createNewItem<Type extends DirectusItem>(
+          {required String typeName,
+          required Map<String, dynamic> objectData,
+          required Type Function(dynamic json) createItemFunction}) {
     return _sendRequest(
         prepareRequest: () =>
             _api.prepareCreateNewItemRequest(typeName, objectData),
-        parseResponse: (response) =>
-            jsonConverter(_api.parseCreateNewItemResponse(response)));
+        parseResponse: (response) {
+          final DirectusItemCreationResult<Type> creationResult =
+              DirectusItemCreationResult(
+                  isSuccess:
+                      response.statusCode == 200 || response.statusCode == 204);
+
+          if (response.statusCode == 200) {
+            creationResult.createdItemList.add(
+                createItemFunction(_api.parseCreateNewItemResponse(response)));
+          }
+          return creationResult;
+        });
   }
 
-  Future<List<Type>> createMultipleItems<Type>(
-      {required String typeName,
-      String fields = "*",
-      required Iterable<Map<String, dynamic>> objectListData,
-      required Type Function(dynamic) jsonConverter}) {
+  Future<DirectusItemCreationResult<Type>>
+      createMultipleItems<Type extends DirectusItem>(
+          {required String typeName,
+          String fields = "*",
+          required Iterable<Map<String, dynamic>> objectListData,
+          required Type Function(dynamic json) createItemFunction}) {
     return _sendRequest(
         prepareRequest: () => _api.prepareCreateNewItemRequest(
             typeName, objectListData,
             fields: fields),
         parseResponse: (response) {
-          final List<Type> createdItemsList = [];
-          final listJson = _api.parseCreateNewItemResponse(response);
-          if (listJson is List) {
-            for (final itemJson in listJson) {
-              createdItemsList.add(jsonConverter(itemJson));
+          final DirectusItemCreationResult<Type> creationResult =
+              DirectusItemCreationResult(
+                  isSuccess:
+                      response.statusCode == 200 || response.statusCode == 204);
+          if (response.statusCode == 200) {
+            final listJson = _api.parseCreateNewItemResponse(response);
+            if (listJson is List) {
+              for (final itemJson in listJson) {
+                creationResult.createdItemList
+                    .add(createItemFunction(itemJson));
+              }
             }
           }
-          return createdItemsList;
+
+          return creationResult;
         });
   }
 
-  Future<Type> updateItem<Type>(
+  Future<Type> updateItem<Type extends DirectusItem>(
       {required String typeName,
       required String objectId,
       required Map<String, dynamic> objectData,
```

Comment:
Shouldn't we pass directly the `Type` object here so we can leverage the `updatedProperties` ourselves like we did for the `updateDirectusUser` ?

Reply:
I meant for the input properties `objectData` and `objectId`, maybe passing in the `Type` object directly would be simpler than sending in the id and the updated properties since they can both be found within the `Type` object.